src/main/java/it/gov/pagopa/payhub/activities/service/OrganizationService.java

```diff
@@ -0,0 +1,31 @@
+package it.gov.pagopa.payhub.activities.service;
+
+import it.gov.pagopa.payhub.activities.dao.OrganizationDao;
+import it.gov.pagopa.payhub.activities.dto.OrganizationDTO;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+/**
+ * Service class exposing methods related to Organization entity.
+ */
+@Lazy
+@Service
+public class OrganizationService {
+
+  private final OrganizationDao organizationDao;
+
+  public OrganizationService(OrganizationDao organizationDao) {
+    this.organizationDao = organizationDao;
+  }
+
+  /*
+   * Retrieve organization entity by fiscal code.
+   * @param orgFiscalCode the fiscal code of the organization to retrieve
+   * @return the retrieved organization
+   */
+  public Optional<OrganizationDTO> getOrganizationByFiscalCode(String orgFiscalCode){
+    return Optional.ofNullable(organizationDao.getOrganizationByFiscalCode(orgFiscalCode));
```

Comment:
we could change the DAO to return an Optional instead of just organizationDTO if we want to handle it always as an optional instead of moving this logic on the Service (SpringData is able to handle Optional, so this behavior would be handled OutOfTheBox by the framework on the new microservices)

Reply:
done